src/history/hs_rec.c

```diff
@@ -1104,7 +1101,11 @@ __hs_delete_record(
     WT_ERR_NOTFOUND_OK(__wt_curhs_search_near_before(session, r->hs_cursor), true);
     /* It's possible the value in the history store becomes obsolete concurrently. */
     if (ret == WT_NOTFOUND) {
-        WT_ASSERT(session, tombstone != NULL && __wt_txn_upd_visible_all(session, tombstone));
+        /* As part of rollback to stable we are adding a globally visible tombstone to unstable
```

Comment:
We need to reformat the comment:
/*
 *
 */

Reply:
I don't understand this 
```
we cannot guarantee there is a tombstone because it is possible for the key to be completely removed from history store after eviction.
```
I thought the key is removed by rts?

Reply:
@quchenhao Good spot, will update the comment. 

It is worded this way because RTS only marks it for deletion by adding a globally visible tombstone - and eviction is the one that removes the key. RTS itself will not remove the key.

Reply:
addressed here: [1789be1](https://github.com/wiredtiger/wiredtiger/pull/9009/commits/1789be12cf0c664b2bead10869a7055e5c640908)